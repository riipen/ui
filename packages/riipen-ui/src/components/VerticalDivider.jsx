import clsx from "clsx";
import PropTypes from "prop-types";
import React from "react";
import _JSXStyle from "styled-jsx/style";

import { withThemeContext } from "../utils";

const VerticalDivider = props => {
  const { classes, spacing, theme, width } = props;

  const className = clsx(classes);

  return (
    <>
      <div className={className} />
      <style jsx>{`
        div {
          background-color: ${theme.palette.divider};
          width: ${width}px;
          margin-left: ${theme.spacing(spacing)}px;
          margin-right: ${theme.spacing(spacing)}px;
        }
      `}</style>
    </>
  );
};

VerticalDivider.propTypes = {
  /**
   * List of additional classes to apply to this component.
   */
  classes: PropTypes.array,

  /**
   * Additional spacing based in theme spacing multiplier to apply after each list item.
   */
  spacing: PropTypes.number,

  /**
   * @ignore
   * The theme context object.
   */
  theme: PropTypes.object,

  /**
   * The width of the divider in pixels.
   */
  width: PropTypes.number
};

VerticalDivider.defaultProps = {
  classes: [],
  spacing: 2,
  width: 1
};

VerticalDivider.displayName = "VerticalDivider";

export default withThemeContext(VerticalDivider);
