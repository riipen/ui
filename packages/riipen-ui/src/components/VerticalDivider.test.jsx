import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import VerticalDivider from "./VerticalDivider";

describe("<VerticalDivider>", () => {
  it("renders without errors", () => {
    let error;

    try {
      mount(<VerticalDivider />);
    } catch (e) {
      error = e;
    }

    expect(error).toEqual(undefined);
  });

  it("renders correct snapshot", () => {
    const wrapper = mount(<VerticalDivider />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("sets correct default props", () => {
    const wrapper = mount(<VerticalDivider />);
    const component = wrapper.find("VerticalDivider");

    expect(component.props().spacing).toEqual(2);
    expect(component.props().width).toEqual(1);
  });

  it("applies custom classes", () => {
    const classes = ["classOne"];

    const wrapper = mount(<VerticalDivider classes={classes} />);

    expect(
      wrapper
        .find("VerticalDivider")
        .childAt(0)
        .hasClass(classes[0])
    ).toEqual(true);
  });
});
