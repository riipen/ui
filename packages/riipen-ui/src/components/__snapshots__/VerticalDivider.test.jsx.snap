// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalDivider> renders correct snapshot 1`] = `
<Component>
  <VerticalDivider
    classes={Array []}
    spacing={2}
    theme={
      Object {
        "breakpoints": Object {
          "lg": 1440,
          "md": 960,
          "sm": 480,
          "xl": 1920,
          "xs": 320,
        },
        "palette": Object {
          "background": Object {
            "default": "#fafafa",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "disabled": "#cccbcb",
          "divider": "#ddd",
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#dddddd",
            "400": "#cccbcb",
            "50": "#fafafa",
            "500": "#9a9a9a",
            "600": "#747474",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A400": "#373737",
          },
          "negative": Object {
            "contrast": "#fff",
            "dark": "#ba2222",
            "light": "#ea7e7e",
            "main": "#e95353",
          },
          "positive": Object {
            "contrast": "#fff",
            "dark": "#1b965c",
            "light": "#78dbac",
            "main": "#4cc88d",
          },
          "primary": Object {
            "contrast": "#fff",
            "dark": "#254364",
            "light": "#2c8ddc",
            "main": "#1f71ae",
          },
          "secondary": Object {
            "contrast": "#fff",
            "dark": "#ed7321",
            "light": "#f5bf95",
            "main": "#f58632",
          },
          "tertiary": Object {
            "contrast": "#fff",
            "dark": "#41978a",
            "light": "#78becc",
            "main": "#3caabb",
          },
          "text": Object {
            "disabled": "#cccbcb",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "#373737",
            "secondary": "#747474",
          },
          "warning": Object {
            "contrast": "#fff",
            "dark": "#cf890e",
            "light": "#ffd894",
            "main": "#febb46",
          },
        },
        "shadows": Object {
          "0": "none",
          "1": "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
          "10": "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "11": "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "12": "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "13": "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "14": "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "15": "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "16": "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "17": "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "18": "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "19": "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "2": "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
          "20": "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "21": "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "22": "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "23": "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "24": "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
          "3": "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
          "4": "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "5": "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "6": "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "7": "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "8": "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "9": "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
        },
        "shape": Object {
          "borderRadius": Object {
            "lg": "6px",
            "md": "4px",
            "sm": "3px",
            "xl": "8px",
          },
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "16px",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": "24px",
            "mobile": Object {
              "fontSize": "14px",
              "lineHeight": "20px",
            },
          },
          "body2": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "14px",
            "fontWeight": 400,
            "letterSpacing": "0.01071em",
            "lineHeight": "20px",
            "mobile": Object {
              "fontSize": "12px",
              "lineHeight": "16px",
            },
          },
          "body3": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "12px",
            "fontWeight": 400,
            "letterSpacing": "0.01071em",
            "lineHeight": "16px",
            "mobile": Object {
              "fontSize": "11px",
              "lineHeight": "16px",
            },
          },
          "button": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "13px",
            "fontWeight": 500,
            "letterSpacing": "0.02857em",
            "lineHeight": "20px",
            "mobile": Object {
              "fontSize": "12px",
              "lineHeight": "16px",
            },
          },
          "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
          "fontWeight": Object {
            "bold": 700,
            "light": 300,
            "medium": 500,
            "regular": 400,
          },
          "h1": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "40px",
            "fontWeight": 500,
            "letterSpacing": "-0.01562em",
            "lineHeight": "56px",
            "mobile": Object {
              "fontSize": "30px",
              "lineHeight": "44px",
            },
          },
          "h2": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "30px",
            "fontWeight": 500,
            "letterSpacing": "-0.00833em",
            "lineHeight": "44px",
            "mobile": Object {
              "fontSize": "22px",
              "lineHeight": "32px",
            },
          },
          "h3": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "22px",
            "fontWeight": 500,
            "letterSpacing": "0em",
            "lineHeight": "32px",
            "mobile": Object {
              "fontSize": "18px",
              "lineHeight": "28px",
            },
          },
          "h4": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "20px",
            "fontWeight": 400,
            "letterSpacing": "-0.00735em",
            "lineHeight": "28px",
            "mobile": Object {
              "fontSize": "16px",
              "lineHeight": "24px",
            },
          },
          "h5": Object {
            "fontFamily": "Roboto, Helvetica, Arial, sans-serif",
            "fontSize": "16px",
            "fontWeight": 400,
            "letterSpacing": "0rem",
            "lineHeight": "24px",
            "mobile": Object {
              "fontSize": "14px",
              "lineHeight": "20px",
            },
          },
        },
        "zIndex": Object {
          "above": 6,
          "high": 4,
          "highest": 5,
          "low": 2,
          "lowest": 1,
          "middle": 3,
          "zero": 0,
        },
      }
    }
    width={1}
  >
    <div
      className="jsx-3601105520 "
    />
    <JSXStyle
      dynamic={
        Array [
          "#ddd",
          1,
          8,
          8,
        ]
      }
      id="3218977506"
    />
  </VerticalDivider>
</Component>
`;
